homeassistant/components/media_player/russound_rnet.py

```diff
@@ -0,0 +1,117 @@
+"""
+
+Support for interfacing with Russound CAV units
+via RNET Protocol.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.russound_rnet/
+"""
+import logging
+
+from homeassistant.components.media_player import (
+    SUPPORT_TURN_OFF, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET,
+    SUPPORT_SELECT_SOURCE, MediaPlayerDevice)
+from homeassistant.const import (
+    CONF_HOST, CONF_PORT, STATE_OFF, STATE_ON)
+
+REQUIREMENTS = [
+    'https://github.com/laf/russound/archive/0.1.6.zip'
+    '#russound==0.1.6']
+
+ZONES = 'zones'
+SOURCES = 'sources'
+
+SUPPORTED_COMMANDS = SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_SET | \
+                     SUPPORT_TURN_OFF | SUPPORT_SELECT_SOURCE
+
+_LOGGER = logging.getLogger(__name__)
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Russound RNET platform."""
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    keypad = config.get('keypad', '70')
+
+    if host is None or port is None:
+        _LOGGER.error('Invalid config. Expected %s and %s',
+                      CONF_HOST, CONF_PORT)
+        return False
+
+    from russound import russound
+
+    russ = russound.Russound(host, port)
+    russ.connect(keypad)
+
+    sources = []
+    for source in config[SOURCES]:
+        sources.append(source['name'])
+
+    if russ.is_connected():
+        for zone_id, extra in config[ZONES].items():
+            add_devices([RussoundRNETDevice(hass, russ, sources, zone_id,
+                                            extra)])
+    else:
+        _LOGGER.error('Not connected to %s:%s', host, port)
+
+# pylint: disable=abstract-method, too-many-public-methods,
+# pylint: disable=too-many-instance-attributes, too-many-arguments
+class RussoundRNETDevice(MediaPlayerDevice):
+    """Representation of a Russound RNET device."""
+    def __init__(self, hass, russ, sources, zone_id, extra):
+        """Initialise the Russound RNET device."""
+        self._name = extra['name']
+        self._russ = russ
+        self._state = STATE_OFF
+        self._sources = sources
+        self._zone_id = zone_id
+
+    @property
+    def name(self):
+        """Return the name of the zone."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def supported_media_commands(self):
+        """Flag of media commands that are supported."""
+        return SUPPORTED_COMMANDS
+
+    @property
+    def volume_level(self):
+        """Volume level of the media player (0..1)."""
+        return 0
```

Comment:
It's possible to adjust outside of HA with things like Keypads or other software. I'll cache the value from set_volume_level for now.

The devices do support requesting for information but the device also spits out statuses every so often, I'm working on trying to make use of that but it's going to take me some time :(